[AppService] Add az webapp|functionapp config ssl create

madsd opened the pull request:
Fixes #11872 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

yonzhan commented:
add to S165.

qwordy commented:
Please add at least one test case

madsd commented:
Unit test added - although it had to be mocked. It is dependent on a CNAME lookup.

Juliehzl commented on src/azure-cli/HISTORY.rst:
```diff
@@ -37,6 +37,7 @@ Release History
 
 * Azure Stack: surface commands under the profile of 2019-03-01-hybrid
 * functionapp: Add ability to create Java function apps in Linux
+* Add az webapp|functionapp config ssl create-managed-certificate
```
Please make your release note in next release part.

madsd replied:
removed the History.rst release note as this is extracted from the title now

Juliehzl commented:
> Unit test added - although it had to be mocked. It is dependent on a CNAME lookup.

Why not use scenario test here?

madsd commented:
> > Unit test added - although it had to be mocked. It is dependent on a CNAME lookup.
> 
> Why not use scenario test here?

As mentioned in our chat - because scenario test will not be able to handle the CNAME registration part that is necessary for a successful run. You will see the same pattern for e.g. setting domain name.

panchagnula commented:
@madsd this feature is not supported only on SKUs Basic & above - so free & shared don't support this can you add a validation for this & throw an error - the API doesn't handle this currently but the FrontEnds block - so we need to handle this on CLI end, we do the same in portal. Thanks!

madsd commented:
> @madsd this feature is not supported only on SKUs Basic & above - so free & shared don't support this can you add a validation for this & throw an error - the API doesn't handle this currently but the FrontEnds block - so we need to handle this on CLI end, we do the same in portal. Thanks!

I added a check for Free and Shared, and will throw a CLI error if it is. I updated the unit tests to handle this.

panchagnula commented on src/azure-cli/azure/cli/command_modules/appservice/commands.py:
```diff
@@ -144,6 +144,7 @@ def load_command_table(self, _):
         g.custom_command('unbind', 'unbind_ssl_cert', validator=validate_app_or_slot_exists_in_rg)
         g.custom_command('delete', 'delete_ssl_cert', exception_handler=ex_handler_factory())
         g.custom_command('import', 'import_ssl_cert', exception_handler=ex_handler_factory(), is_preview=True)
+        g.custom_command('create-managed-certificate', 'create_managed_ssl_cert', exception_handler=ex_handler_factory(), is_preview=True)
```
@yonzhan / @Juliehzl can you comment if this naming follows the recommended naming convention
@madsd  i was thinking if we should call this create simply & have another property that defaults to Managed certs as default for now so that we can re-purpose the same command to create AppService certificates when we are ready to add that support as well? What do you think?  If you agree , we can run this by Yutang as well as the PM for this feature.

panchagnula replied:
@yutanglin16 - FYI
